src/v/redpanda/application.cc

```diff
@@ -275,6 +277,15 @@ void application::initialize(
           config::shard_local_cfg().zstd_decompress_workspace_bytes());
     }).get0();
 
+    /*
+     * allocate per-core workspaces for async_stream_zstd. it can be several
+     * megabytes in size, so do it before memory becomes fragmented.
+     */
+    ss::smp::invoke_on_all([] {
+        compression::async_stream_zstd::init_workspace(
+          config::shard_local_cfg().zstd_decompress_workspace_bytes());
+    }).get0();
```

Comment:
is this a workspace that is now separate from the non-async version? should one be replaced or can they share? if they are separate what's the future of them in terms of being combined?

Reply:
It is separated since they can't be shared without a mutex. Though I suppose we could have a compress/decompress mutex both classes share. The hope is that we can replace `stream_zstd` with `async_stream_zstd` in `v/kafka` as which point we can remove `stream_zstd`.

Reply:
> The hope is that we can replace stream_zstd with async_stream_zstd in v/kafka as which point we can remove stream_zstd.

yeh i think this makes sense as the preferred path forward. assuming we don't find any issues with performance we can probably make that switch soon